parsers/jscript.c

```diff
@@ -46,6 +47,52 @@
 #define deleteToken(t) (objPoolPut (TokenPool, (t)))
 
 /*
+ * Debugging
+ *
+ * Uncomment this to enable extensive debugging to stderr in jscript code.
+ * Please note that TRACING_ENABLED should be #defined in main/trace.h
+ * for this to work.
+ *
+ */
+//#define JSCRIPT_DEBUGGING_ENABLED 1
+
+#if defined(DO_TRACING) && defined(JSCRIPT_DEBUGGING_ENABLED)
+	#define JSCRIPT_DO_DEBUGGING
+#endif
+
+#ifdef JSCRIPT_DO_DEBUGGING
+
+#define JSCRIPT_DEBUG_ENTER() TRACE_ENTER()
+#define JSCRIPT_DEBUG_LEAVE() TRACE_LEAVE()
+
+#define JSCRIPT_DEBUG_ENTER_TEXT(_szFormat,...) \
+	TRACE_ENTER_TEXT(_szFormat,## __VA_ARGS__)
+
+#define JSCRIPT_DEBUG_LEAVE_TEXT(_szFormat,...) \
+	TRACE_LEAVE_TEXT(_szFormat,## __VA_ARGS__)
+
+#define JSCRIPT_DEBUG_PRINT(_szFormat,...) \
+	TRACE_PRINT(_szFormat,## __VA_ARGS__)
+
+#define JSCRIPT_DEBUG_ASSERT(_condition,_szFormat,...) \
+	TRACE_ASSERT(_condition,_szFormat,## __VA_ARGS__)
+
+#else //!JSCRIPT_DO_DEBUGGING
+
+#define JSCRIPT_DEBUG_ENTER() do { } while(0)
+#define JSCRIPT_DEBUG_LEAVE() do { } while(0)
+
+#define JSCRIPT_DEBUG_ENTER_TEXT(_szFormat,...) do { } while(0)
+#define JSCRIPT_DEBUG_LEAVE_TEXT(_szFormat,...) do { } while(0)
+
+#define JSCRIPT_DEBUG_PRINT(_szFormat,...) do { } while(0)
+
+#define JSCRIPT_DEBUG_ASSERT(_condition,_szFormat,...) do { } while(0)
+
+#endif //!JSCRIPT_DO_DEBUGGING
+
+
+/*
```

Comment:
Do you think this layer is really needed?
I think using TRACE_xxx directly is o.k.

@b4n, how do you think?

Even if we decide to use the prefix, I have a comment.
This change, adding JSCRIPT_ prefixed macros should be in a separated commit.